### PR TITLE
FOHM upload existing GISAID ID

### PR DESCRIPTION
**Pull request by Mropat**
### Added

-

### Changed

-

### Fixed

-

### How to prepare for test
- [ ] ssh to relevant server (depending on type of change)
- [ ] Use stage: `us`
- [ ] paxa the environment: `paxa`
- [ ] install on stage (example for hasta):
`bash /home/proj/production/servers/resources/hasta.scilifelab.se/update-cg-stage.sh [THIS-BRANCH-NAME]`

### How to test
- [ ] do ...

### Expected test outcome
- [ ] check that ...
- [ ] Take a screenshot and attach or copy/paste the output.

## Review
- [ ] code approved by
- [ ] tests executed by
- [ ] "Merge and deploy" approved by
Thanks for filling in who performed the code review and the test!

### This [version](https://semver.org/) is a
- [ ] **MAJOR** - when you make incompatible API changes
- [ ] **MINOR** - when you add functionality in a backwards compatible manner
- [ ] **PATCH** - when you make backwards compatible bug fixes or documentation/instructions

## Implementation Plan
- [ ] Document in ...
- [ ] Deploy this branch
- [ ] Inform to ...

### CODE REVIEW

File: cg/meta/upload/gisaid/gisaid.py [hunk 1]
```diff
@@ -23,6 +24,8 @@
 
 LOG = logging.getLogger(__name__)
 
+UPLOADED_REGEX_MATCH = r"\[\"([A-Za-z0-9_]+)\"\]\}$"
```
Comment (henningonsbring) [inline cg/meta/upload/gisaid/gisaid.py:27]: 🤑

File: cg/meta/upload/gisaid/gisaid.py [hunk 2]
```diff
@@ -276,9 +279,16 @@ def get_accession_numbers(self, case_id: str) -> Dict[str, str]:
             with open(str(log_file.absolute())) as log_file:
                 log_data = json.load(log_file)
                 for log in log_data:
-                    if log.get("code") != "epi_isl_id":
+                    if log.get("code") == "epi_isl_id":
+                        log_message = log.get("msg")
+                    elif log.get("code") == "validation_error" and "existing_ids" in log.get("msg"):
+                        log_message = (
+                            f'{log.get("msg").split(";")[0]}; '
```
Comment (henningonsbring) [inline cg/meta/upload/gisaid/gisaid.py:286]: Is this correct with splitting on ; then looking at [0]? I guess it is correct if it works, but I do not understand this part

**Comment by Mropat** [comment]
Tested locally since we have no GISAID dry-run:


```
UPLOADED_REGEX_MATCH = r"\[\"([A-Za-z0-9_]+)\"\]\}$"
with open(file_path) as log_file:
    log_data = json.load(log_file)
    for log in log_data:
        if log.get("code") == "epi_isl_id":
            log_message = log.get("msg")
        elif log.get("code") == "validation_error" and "existing_ids" in log.get("msg"):
            log_message = (
                f'{log.get("msg").split(";")[0]}; '
                f'{re.findall(UPLOADED_REGEX_MATCH, log.get("msg"))[0]}'
            )
        print(log_message)      
```

acieves same formatting for hCoV-19/Sweden/01_SE100_21CS106873/2021
```
hCoV-19/Sweden/09_SE100_21CS106842/2021; EPI_ISL_7486453
hCoV-19/Sweden/01_SE100_21CS106873/2021; EPI_ISL_7286284
hCoV-19/Sweden/01_SE100_21CS106861/2021; EPI_ISL_7486454
hCoV-19/Sweden/01_SE100_21CS106862/2021; EPI_ISL_7486455
hCoV-19/Sweden/01_SE100_21CS106831/2021; EPI_ISL_7486456
hCoV-19/Sweden/01_SE100_21CS106859/2021; EPI_ISL_7486457
hCoV-19/Sweden/09_SE100_21CS106854/2021; EPI_ISL_7486458
hCoV-19/Sweden/01_SE100_21CS106887/2021; EPI_ISL_7486459
hCoV-19/Sweden/09_SE100_21CS106850/2021; EPI_ISL_7486460
hCoV-19/Sweden/01_SE100_21CS106874/2021; EPI_ISL_7486461
hCoV-19/Sweden/01_SE100_21CS106883/2021; EPI_ISL_7486462
hCoV-19/Sweden/01_SE100_21CS106875/2021; EPI_ISL_7486463
hCoV-19/Sweden/01_SE100_21CS106824/2021; EPI_ISL_7486464
hCoV-19/Sweden/01_SE100_21CS106870/2021; EPI_ISL_7486465
hCoV-19/Sweden/01_SE100_21CS106799/2021; EPI_ISL_7486466
hCoV-19/Sweden/01_SE100_21CS106872/2021; EPI_ISL_7486467
hCoV-19/Sweden/01_SE100_21CS106826/2021; EPI_ISL_7486468
hCoV-19/Sweden/01_SE100_21CS106889/2021; EPI_ISL_7486469
hCoV-19/Sweden/01_SE100_21CS106869/2021; EPI_ISL_7486470
hCoV-19/Sweden/01_SE100_21CS106879/2021; EPI_ISL_7486471
hCoV-19/Sweden/01_SE100_21CS106838/2021; EPI_ISL_7486472
hCoV-19/Sweden/01_SE100_21CS106865/2021; EPI_ISL_7486478
hCoV-19/Sweden/01_SE100_21CS106877/2021; EPI_ISL_7486486
hCoV-19/Sweden/09_SE100_21CS106849/2021; EPI_ISL_7486497
hCoV-19/Sweden/01_SE100_21CS106830/2021; EPI_ISL_7486504
hCoV-19/Sweden/01_SE100_21CS106801/2021; EPI_ISL_7486513
hCoV-19/Sweden/09_SE100_21CS106856/2021; EPI_ISL_7486519
hCoV-19/Sweden/01_SE100_21CS106884/2021; EPI_ISL_7486526
hCoV-19/Sweden/01_SE100_21CS106800/2021; EPI_ISL_7486530
hCoV-19/Sweden/01_SE100_21CS106866/2021; EPI_ISL_7486534
hCoV-19/Sweden/09_SE100_21CS106845/2021; EPI_ISL_7486542
hCoV-19/Sweden/01_SE100_21CS106816/2021; EPI_ISL_7486549
hCoV-19/Sweden/01_SE100_21CS106798/2021; EPI_ISL_7486559
hCoV-19/Sweden/01_SE100_21CS106881/2021; EPI_ISL_7486569
hCoV-19/Sweden/01_SE100_21CS106815/2021; EPI_ISL_7486574
hCoV-19/Sweden/01_SE100_21CS106858/2021; EPI_ISL_7486581
hCoV-19/Sweden/01_SE100_21CS106863/2021; EPI_ISL_7486591
hCoV-19/Sweden/01_SE100_21CS106890/2021; EPI_ISL_7486603
hCoV-19/Sweden/01_SE100_21CS106805/2021; EPI_ISL_7486612
hCoV-19/Sweden/01_SE100_21CS106835/2021; EPI_ISL_7486616
hCoV-19/Sweden/01_SE100_21CS106810/2021; EPI_ISL_7486625
hCoV-19/Sweden/01_SE100_21CS106818/2021; EPI_ISL_7486632
hCoV-19/Sweden/01_SE100_21CS106833/2021; EPI_ISL_7486638
hCoV-19/Sweden/01_SE100_21CS106876/2021; EPI_ISL_7486642
hCoV-19/Sweden/01_SE100_21CS106857/2021; EPI_ISL_7486650
hCoV-19/Sweden/01_SE100_21CS106839/2021; EPI_ISL_7486657
hCoV-19/Sweden/01_SE100_21CS106812/2021; EPI_ISL_7486664
hCoV-19/Sweden/01_SE100_21CS106817/2021; EPI_ISL_7486670
hCoV-19/Sweden/01_SE100_21CS106840/2021; EPI_ISL_7486673
hCoV-19/Sweden/01_SE100_21CS106803/2021; EPI_ISL_7486678
hCoV-19/Sweden/01_SE100_21CS106827/2021; EPI_ISL_7486686
hCoV-19/Sweden/01_SE100_21CS106809/2021; EPI_ISL_7486692
hCoV-19/Sweden/01_SE100_21CS106802/2021; EPI_ISL_7486700
hCoV-19/Sweden/01_SE100_21CS106880/2021; EPI_ISL_7486707
hCoV-19/Sweden/01_SE100_21CS106888/2021; EPI_ISL_7486717
hCoV-19/Sweden/01_SE100_21CS106868/2021; EPI_ISL_7486724
hCoV-19/Sweden/01_SE100_21CS106811/2021; EPI_ISL_7486732
hCoV-19/Sweden/01_SE100_21CS106886/2021; EPI_ISL_7486736
hCoV-19/Sweden/09_SE100_21CS106843/2021; EPI_ISL_7486743
hCoV-19/Sweden/01_SE100_21CS106832/2021; EPI_ISL_7486752
hCoV-19/Sweden/01_SE100_21CS106871/2021; EPI_ISL_7486756
hCoV-19/Sweden/01_SE100_21CS106860/2021; EPI_ISL_7486760
hCoV-19/Sweden/01_SE100_21CS106885/2021; EPI_ISL_7486767
hCoV-19/Sweden/01_SE100_21CS106814/2021; EPI_ISL_7486774
hCoV-19/Sweden/09_SE100_21CS106848/2021; EPI_ISL_7486779
hCoV-19/Sweden/01_SE100_21CS106878/2021; EPI_ISL_7486787
hCoV-19/Sweden/01_SE100_21CS106808/2021; EPI_ISL_7486798
hCoV-19/Sweden/01_SE100_21CS106864/2021; EPI_ISL_7486803
hCoV-19/Sweden/01_SE100_21CS106864/2021; EPI_ISL_7486803
hCoV-19/Sweden/01_SE100_21CS106864/2021; EPI_ISL_7486803

```

File: cg/meta/upload/gisaid/gisaid.py (hunk 2 above)
Comment (Mropat) [reply cg/meta/upload/gisaid/gisaid.py:286]: yes it works - also thats what we do for successful uploads aswell

**Review by henningonsbring — APPROVED** [review]
lets merge

**Comment by Mropat** [comment]
```
Logging deploy ...
Getting deployer... done.
Getting last commit message and SHA... done.
Getting version of deploy scripts... /home/maryia.ropart
done.
Log deploy... done.
Running --version on each Clinical-Genomics developed tool to verify that all requirements are met:
balsamic 6.0.0 requires colorclass, which is not installed.
argcomplete 1.12.1 has requirement importlib-metadata<3,>=0.23, but you have importlib-metadata 0.8.
cg, version 22.36.1
housekeeper, version 3.2.0
demux, version 5.10.3
cglims, version 1.3.4
```

**Pull request merged**